classes/Address.php

```diff
@@ -520,8 +520,9 @@ public static function aliasExist($alias, $id_address, $id_customer)
     public function getFieldsRequiredDB()
     {
         $this->cacheFieldsRequiredDatabase(false);
-        if (isset(self::$fieldsRequiredDatabase['Address'])) {
-            return self::$fieldsRequiredDatabase['Address'];
+        $objectName = $this->getObjectName();
+        if (isset(self::$fieldsRequiredDatabase[$objectName])) {
+            return self::$fieldsRequiredDatabase[$objectName];
```

Comment:
This function only use methods and attributes from the parent class => should move it to ObjectModel.
+ code duplication in ObjectModel